logic/src/commonTest/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserRepositoryTest.kt

```diff
@@ -0,0 +1,324 @@
+                .whenInvokedWith(any())
+                .then { TestNetworkResponseError.genericError() }
+            //when
+            publicUserRepository.searchPublicContact("test")
+
+            //then
+            verify(contactSearchApi)
+                .suspendFunction(contactSearchApi::search)
+                .with(any())
+                .wasInvoked(exactly = once)
+
+            verify(userDetailsApi)
+                .suspendFunction(userDetailsApi::getMultipleUsers)
+                .with(any())
+                .wasInvoked(exactly = once)
+        }
+
+    @Test
+    fun givenContactSearchApiAndUserDetailsApiAndPublicUserApiReturnSuccess_WhenSearchPublicContact_ThenResultIsSuccess() = runTest {
+        //given
+        given(contactSearchApi)
+            .suspendFunction(contactSearchApi::search)
+            .whenInvokedWith(any())
+            .then { NetworkResponse.Success(CONTACT_SEARCH_RESPONSE, mapOf(), 200) }
+
+        given(userDetailsApi)
+            .suspendFunction(userDetailsApi::getMultipleUsers)
+            .whenInvokedWith(any())
+            .then { NetworkResponse.Success(GET_MULTIPLE_USER_RESPONSE, mapOf(), 200) }
+
+        given(publicUserMapper)
+            .function(publicUserMapper::fromUserDetailResponses)
+            .whenInvokedWith(any())
+            .then { PUBLIC_USERS }
+
+        //when
+        val actual = publicUserRepository.searchPublicContact("test")
+
+        //then
+        assertIs<Either.Right<PublicUserSearchResult>>(actual)
+    }
+
+    @Test
+    fun givenContactSearchApiAndUserDetailsApiAndPublicUserApiReturnSuccess_WhenSearchPublicContact_ThenResultIsEqualToExpectedValue() =
+        runTest {
+            //given
+            given(contactSearchApi)
+                .suspendFunction(contactSearchApi::search)
+                .whenInvokedWith(any())
+                .then { NetworkResponse.Success(CONTACT_SEARCH_RESPONSE, mapOf(), 200) }
+
+            given(userDetailsApi)
+                .suspendFunction(userDetailsApi::getMultipleUsers)
+                .whenInvokedWith(any())
+                .then { NetworkResponse.Success(GET_MULTIPLE_USER_RESPONSE, mapOf(), 200) }
+
+            given(publicUserMapper)
+                .function(publicUserMapper::fromUserDetailResponses)
+                .whenInvokedWith(any())
+                .then { PUBLIC_USERS }
+
+            val expectedResult = PublicUserSearchResult(
+                totalFound = PUBLIC_USERS.size,
+                publicUsers = PUBLIC_USERS
+            )
+            //when
+            val actual = publicUserRepository.searchPublicContact("test")
+
+            assertIs<Either.Right<PublicUserSearchResult>>(actual)
+            assertEquals(expectedResult, actual.value)
+        }
+
+    @Test
+    fun givenContactSearchApiAndUserDetailsApiAndPublicUserApiReturnSuccessWithNoResult_WhenSearchPublicContact_ThenResultIsEqualToEmptyList() =
```

Comment:
I'm not sure how useful is the way we title these tests, if the title simply states what the test does... Wouldn't it be a bit more readable if we wrote the titles in a more natural language approach?
```suggestion
    fun givenAValidUserSearchWithEmptyResults_WhenSearchingSomeText_ThenResultIsAnEmptyList() =
```

Reply:
totally agrree, thanks